pkg/driver/driver.go

```diff
@@ -967,6 +976,14 @@ func (d Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotReque
 		return nil, status.Errorf(codes.Internal, "failed to create snapshot: %v", err)
 	}
 
+	if snap.ReadyToUse && params.Type != volume.SnapshotTypeInCluster && params.DeleteLocal {
```

Comment:
this whole block is basically a copy from above. please make that a function; at least the check.

Reply:
Good point, moved to a function and cleaned it up a bit.